bridgenode/flatfileworker.go

```diff
@@ -312,23 +325,45 @@ func (ff *flatFileState) writeProofBlock(ud btcacc.UData) error {
 
 func (ff *flatFileState) writeTTLs(ttlRes ttlResultBlock) error {
 	var ttlArr [4]byte
+	var buffer [8]byte
+
+	// write offset of current TTL to offset file
+	ff.offsets = append(ff.offsets, ff.currentOffset)
+
+	// write ttl offset to offsetfile
+	binary.BigEndian.PutUint64(buffer[:], uint64(ff.currentOffset))
+	_, err := ff.offsetFile.WriteAt(buffer[:], int64(8*ttlRes.Height))
+	if err != nil {
+		return err
+	}
+
 	// for all the TTLs, seek and overwrite the empty values there
 	for _, c := range ttlRes.Created {
 		// seek to the location of that txo's ttl value in the proof file
 
 		binary.BigEndian.PutUint32(
 			ttlArr[:], uint32(ttlRes.Height-c.createHeight))
+		// for each iteration of  append 0x00
```

Comment:
This code doesn't work since there should be space allocated *for every leaf*.

Try inserting this snippet at line 354 (also change line 120 to  `utreeDir.TtlDir.ttlsetFile, os.O_CREATE|os.O_RDWR, 0600)`

```
                 shouldBeZeros := make([]byte, 4)                                                                                                                                                                                              
                 _, err = ff.proofFile.ReadAt(shouldBeZeros, ff.offsets[c.createHeight]+int64(c.indexWithinBlock*4))                                                                                                                           
                 if err != nil {                                                                                                                                                                                                               
                         return err                                                                                                                                                                                                            
                 }                                                                                                                                                                                                                             
                 if !bytes.Equal(shouldBeZeros, []byte{0x00, 0x00, 0x00, 0x00}) {                                                                                                                                                              
                         err := fmt.Errorf("read %x, should have read all zeros", shouldBeZeros)                                                                                                                                               
                         panic(err)                                                                                                                                                                                                            
                 }
```

This will fail and say that you read some none 0s.

Per every leaf created, we need to allocate 0s on disk. This means that we need to allocate when we generate the leaf to add. (found here:)

https://github.com/mit-dci/utreexo/blob/5f26803e7056a7a8c1bae3414f1711f37518c3ba/bridgenode/genproofs.go#L145

So for each len(blockAdds), we need to

```
bytesToWrite := make([]byte, len(blockAdds))
_, err := ff.proofFile.WriteAt(bytesToWrite, ff.currentOffset)
if err != nil {
    return err
}
```

I think the best way to give this information to flatFileWorker may just be to have another channel for just sending over the blockAdds and having another function that handles just this allocation on disk.